client/config/config.go

```diff
@@ -203,6 +203,14 @@ type Config struct {
 	// allocation metrics to remote Telemetry sinks
 	PublishAllocationMetrics bool
 
+	// IncludeMetadataInMetrics determines whether nomad should include the
+	// metadata as labels in the metrics to remote Telemetry sinks
```

Comment:
We should be really specific here and call out this is job/allocation metadata, as opposed to client metadata. But that makes me think we should probably make this configuration `IncludeAllocMetadataInMetrics` so that we have the option to add client metadata later if we want.